Mage.Tests/src/test/java/org/mage/test/cards/conditional/SentinelSarahLyonsTest.java

```diff
@@ -0,0 +1,91 @@
+package org.mage.test.cards.conditional;
+
+import mage.constants.PhaseStep;
+import mage.constants.Zone;
+import org.junit.Test;
+import org.mage.test.serverside.base.CardTestPlayerBase;
+
+public class SentinelSarahLyonsTest extends CardTestPlayerBase {
+
+    /*
+     * Sentinel Sarah Lyons {3}{R}{W}
+     * Legendary Creature - Human Knight (4/4)
+     * Haste
+     * As long as an artifact entered the battlefield under your control this turn, creatures you control get +2/+2.
+     * Battalion - Whenever Sentinel Sarah Lyons and at least two other creatures attack,
+     * Sentinel Sarah Lyons deals damage equal to the number of artifacts you control to target player.
+     */
+
+    @Test
+    public void test_ArtifactEntersUnderYourControlBeforeAndAfter() {
+        addCard(Zone.BATTLEFIELD, playerA, "Sentinel Sarah Lyons", 1);
+        addCard(Zone.HAND, playerA, "Darksteel Relic", 1); // Artifact w/ Indestructible
+
+        // Before casting an artifact (4/4)
+        setStrictChooseMode(true);
+        setStopAt(1, PhaseStep.UPKEEP);
+        execute();
```

Comment:
Multiple `execute` in the same test has some weird issues with watcher being reset for instance. 

I would rather have `assertPowerToughness("1: before Casting Relic", 1, PhaseStep.UPKEEP, playerA, "Sentinel Sarah Lyons", 4, 4)`, instead of `setStopAt`/`execute`/`assertPowerToughness`.

Most of the `assert*` methods have an equivalent `check*` that can work on a live game.

Reply:
Good idea, thank you.

Interestingly, the final assertion is failing now. It seems that Sarah Lyons is still a 6/6 during the next turn's untap step for some reason, although it's a 4/4 once we go to upkeep. I observed this behavior in game too.

Reply:
it will be a 6/6 until the first continuous Effect computation on the new turn, as the effect expire between turns.
I don't mind the execute until turn 2 `UPKEEP` there. If you still think there is something weird with the first continuous effect computation, please open a new issue. This is a complex part of the engine, and unrelated with current PR.

Reply:
Thank you. I figured it was probably fine since there's no priority in the untap step, but I thought I should mention it just in case. I'll switch to checking on `UPKEEP`.